app/change_sets/ephemera_box_change_set.rb

```diff
@@ -18,19 +19,26 @@ class EphemeraBoxChangeSet < Valhalla::ChangeSet
   property :rights_note, multiple: false, required: false
   delegate :human_readable_type, to: :model
   validate :barcode_valid?
+  validate :drive_barcode_valid?
 
   def barcode_valid?
     return if Barcode.new(Array.wrap(barcode).first).valid?
     errors.add(:barcode, 'has an invalid checkdigit')
   end
 
+  def drive_barcode_valid?
+    return if drive_barcode.nil? || drive_barcode.empty? || Barcode.new(Array.wrap(drive_barcode).first).valid?
```

Comment:
Shouldn't `Barcode` return a `false` for `#valid?` if a `nil` or empty value is supplied to the constructor?

Please see https://github.com/pulibrary/figgy/blob/master/app/values/barcode.rb#L9

Reply:
Yes, it would — but the nil and empty checks prevent the test from getting there.  This is to allow empty drive barcodes (since there's no requirement for a drive barcode to be supplied).